<a href="https://colab.research.google.com/github/EkaterinaGugina/PythonJupyter_analyz/blob/main/Final_Project_Python_Jupyter.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# HOMEWORK_11_11_2022

**ЗАДАНИЕ: Провести анализ, найти закономерности и сделать вывод по данным файла train (1).csv.**

Каждая строчка наборов данных содержит следующие поля:

* Survived — выжил данный пассажир или нет (0 для умерших, 1 для выживших)

* Pclass — класс пассажира (1 — высший, 2 — средний, 3 — низший);

* Name — имя;

* Sex — пол;

* Age — возраст;

* SibSp — количество братьев, сестер, сводных братьев, сводных сестер, супругов на борту титаника;

* Parch — количество родителей, детей (в том числе приемных) на борту Титаника;

* Ticket — номер билета;

* Fare — плата за проезд;

* Cabin — каюта;

* Embarked — порт посадки (C — Шербур; Q — Квинстаун; S — Саутгемптон).



---


1. Загружаем данные и выводим для просмотра:

In [1]:
import pandas as pd
df = pd.read_csv('train (1).csv')
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]



---


2. Выведем размеры таблицы с данными

In [2]:
df.shape

(891, 12)

Итак, у нас имеется для анализа 12 колонок и 891 строка



---


3. Изучим общую информацию о данных в таблице:

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Видим, что есть три колонки (Age, Cabin, Embarked), данных в которых меньше, чем 891, т.е. некоторые строки не заполнены. Это можно объяснить, нпример, так:

1) у 177 пассажиров нет данных о возрасте, т.к. для покупки билетов в прошлом веке не требовался паспорт? или свиделеьство о рождении и данные заполнялись после катастрофы. Если чей то возраст был неизвестен, то и получили пустое значение;

2) у двоих порт посадки не заполнен данными, что по-видимому является случайным пропуском;

3) каюта не отмечена у *большого* числа людей - у 687. Видимо эти данные не являлись основными, не сохранились и заполнены только у тех выживших, кто назвал номер своей каюты.

       Проверим поэтому следующие гипотезы, подверждающие или опровергающие наши предположения:

* выведем сначала общее количество выживших и погибших, может номер каюты не смогли назвать погибшие?;

* выясним, отмечен ли возраст у всех членов семьи для пассажиров, путешествовавших со своими родственниками. Для этого ВЫВЕДЕМ СНАЧАЛА  ПАССАЖИРОВ, ПУТЕШЕСТВОВАВШИХ СЕМЬЕЙ, с колонкой данных об их спасении и возраста. Может, все семейные выжившие назвали свой возраст, а погибшие не смогли этого сделать, отсюда потеря данных;

* ;

* 



---


Проанализируем количество выживших и погибших пассажиров:

In [4]:
df.Survived.value_counts()

0    549
1    342
Name: Survived, dtype: int64

Судя, по количеству погибших = 549, меньшему 687 - числа незаполненных данных по каюте, номер каюты был неизвестен, в том числе, не у всех выживших. Т.е. предполагаем, что не все были с каютами?, плыли в общих местах пользования (палубы?).



---



Выведем из основной таблицы только четыре колонки: 'Name', 'Age', 'SibSp',  'Survived':

In [37]:
df_1 = df[['Name', 'SibSp', 'Age', 'Survived']]
df_1

Name  SibSp   Age  Survived
0                              Braund, Mr. Owen Harris      1  22.0         0
1    Cumings, Mrs. John Bradley (Florence Briggs Th...      1  38.0         1
2                               Heikkinen, Miss. Laina      0  26.0         1
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)      1  35.0         1
4                             Allen, Mr. William Henry      0  35.0         0
..                                                 ...    ...   ...       ...
886                              Montvila, Rev. Juozas      0  27.0         0
887                       Graham, Miss. Margaret Edith      0  19.0         1
888           Johnston, Miss. Catherine Helen "Carrie"      1   NaN         0
889                              Behr, Mr. Karl Howell      0  26.0         1
890                                Dooley, Mr. Patrick      0  32.0         0

[891 rows x 4 columns]



---


Выведем среди них пассажиров, путешествующих семьей. Затем, если среди них есть пассажиры без данных о возрасте, найдем их количество:

In [52]:
df_family = df_1.loc[df_1['SibSp'] == 1]
df_family

Name  SibSp   Age  Survived
0                              Braund, Mr. Owen Harris      1  22.0         0
1    Cumings, Mrs. John Bradley (Florence Briggs Th...      1  38.0         1
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)      1  35.0         1
9                  Nasser, Mrs. Nicholas (Adele Achem)      1  14.0         1
10                     Sandstrom, Miss. Marguerite Rut      1   4.0         1
..                                                 ...    ...   ...       ...
866                       Duran y More, Miss. Asuncion      1  27.0         1
869                    Johnson, Master. Harold Theodor      1   4.0         1
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)      1  47.0         1
874              Abelson, Mrs. Samuel (Hannah Wizosky)      1  28.0         1
888           Johnston, Miss. Catherine Helen "Carrie"      1   NaN         0

[209 rows x 4 columns]

Это 209 пассажиров.

In [59]:
nulls = pd.isnull(df_family['Age']).sum()
nulls

26

Таким образом, среди семейных пассажиров 26 человек были без указания возраста. 

Найдем среди пассажиров "без возраста" количество уцелевших.

In [ ]:
df_family = df_1.loc[df_1['SibSp'] == 1]
df_family
df.Survived.value_counts()

In [ ]:
df_family.loc[df_family['Age'] == 'NaN']
df_familyNotAge = df_family.loc[df_famaly['Age'] == 'NaN']
df_familyNotAge

In [ ]:
df_family.pivot_table(['Survived'],['Name'], aggfunc='sum', fill_value = 0)
df_familyY = df_family[df_family.SibSp == '1']
df_familyY
df_family['SibSib' == '1']